nutils/numeric.py

```diff
@@ -50,6 +50,16 @@ def overlapping(arr, axis=-1, n=2):
   overlapping.flags.writeable = False
   return overlapping
 
+def full(shape, fill_value, dtype):
+  'read-only equivalent to :func:`numpy.full`'
+
+  z = (0,)*len(shape)
+  f = numpy.ndarray(shape=shape, strides=z, dtype=dtype)
```

Comment:
Could use `f = numpy.broadcast_to(fill_value, shape, dtype)` here.

Reply:
The point here is `strides`, which allows us to blow up a scalar to any shape without allocating more memory.